src/eko/io/struct.py

```diff
@@ -0,0 +1,1081 @@
+    @property
+    def theory_card(self):
+        """Provide theory card, retrieving from the dump."""
+        return TheoryCard.from_dict(
+            yaml.safe_load(self.paths.theory_card.read_text(encoding="utf-8"))
+        )
+
+    @property
+    def operator_card(self):
+        """Provide operator card, retrieving from the dump."""
+        return OperatorCard.from_dict(
+            yaml.safe_load(self.paths.operator_card.read_text(encoding="utf-8"))
+        )
+
+    # persistency control
+    # -------------------
+
+    def update(self):
+        """Write updates to structure for persistency."""
+        self.access.assert_writeable()
+        self.metadata.update()
+
+    def assert_permissions(self, read=True, write=False):
+        """Assert permissions on current operator."""
+        if read:
+            self.access.assert_open()
+        if write:
+            self.access.assert_writeable()
+
+    @property
+    def permissions(self):
+        """Provide permissions information."""
+        return dict(read=self.access.read, write=self.access.write)
+
+    # operator management
+    # -------------------
+    def __getitem__(self, mu2: float) -> Operator:
+        r"""Retrieve operator for given :math:`\mu^2`.
+
+        If the operator is not already in memory, it will be automatically
+        loaded.
+
+        Parameters
+        ----------
+        mu2 : float
+            :math:`\mu^2` value labeling the operator to be retrieved
+
+        Returns
+        -------
+        Operator
+            the retrieved operator
+
+        """
+        self.access.assert_open()
+
+        if mu2 in self._operators:
+            op = self._operators[mu2]
+            if op is not None:
+                return op
+
+        oppath = self.paths.oppath(mu2)
+        compressed = self.paths.opcompressed(oppath)
+
+        with open(oppath, "rb") as fd:
+            op = Operator.load(fd, compressed=compressed)
+
+        self._operators[mu2] = op
+        return op
+
+    def __setitem__(self, mu2: float, op: Operator, compress: bool = True):
+        """Set operator for given :math:`Q^2`.
+
+        The operator is automatically dumped on disk, .
+
+        Parameters
+        ----------
+        q2 : float
+            :math:`Q^2` value labeling the operator to be set
+        op : Operator
+            the retrieved operator
+        compress : bool
+            whether to save the operator compressed or not (default: `True`)
+
+        """
+        self.access.assert_writeable()
+
+        if not isinstance(op, Operator):
+            raise ValueError("Only an Operator can be added to an EKO")
+
+        without_err = op.error is None
+        oppath = self.paths.opnewpath(mu2, compress=compress, without_err=without_err)
+
+        with open(oppath, "wb") as fd:
+            without_err2 = op.save(fd, compress)
+        assert without_err == without_err2
+
+        self._operators[mu2] = op
+
+    def __delitem__(self, q2: float):
```

Comment:
The property not, but the operator yes.

You might be right, but this is more intuitive on another side, since:
```py
del eko[10.0]
```
is more intuitively destroying an operator rather than:
```py
eko[10.0] = None
```

I understand it is confusing with respect to usual dictionary syntax, but it is somehow the analogue of `pathlib` usage of `__truediv__`: it is not a division, but a concatenation. Intuitively it is closer to `+`, but in that context the symbol makes sense.

Reply:
In any case, usually you are not supposed to use this directly, there are better user facing methods (like the `.items()` iterator or the `.operator()` context manager).